x/airdrop/keeper/claim_test.go

```diff
@@ -0,0 +1,450 @@
+package keeper_test
+
+import (
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Stride-Labs/stride/v22/x/airdrop/types"
+)
+
+// Helper function to cast an array of allocations as int64's into sdkmath.Ints
+func allocationsToSdkInt(allocationsInt64 []int64) (allocationsSdkInt []sdkmath.Int) {
+	for _, allocation := range allocationsInt64 {
+		allocationsSdkInt = append(allocationsSdkInt, sdkmath.NewInt(allocation))
+	}
+	return allocationsSdkInt
+}
+
+// Helper function to cast an array of allocations as sdk.Int's into int64
+func allocationsToInt64(allocationsSdkInt []sdkmath.Int) (allocationsInt64 []int64) {
+	for _, allocation := range allocationsSdkInt {
+		allocationsInt64 = append(allocationsInt64, allocation.Int64())
+	}
+	return allocationsInt64
+}
+
+func (s *KeeperTestSuite) TestClaimDaily() {
+	testCases := []struct {
+		name                string
+		timeOffset          time.Duration
+		initialAllocations  []int64
+		expectedAllocations []int64
+		initialForfeited    int64
+		expectedForfeited   int64
+		initialClaimed      int64
+		expectedClaimed     int64
+		expectedNewRewards  int64
+		initialClaimType    types.ClaimType
+		expectedError       string
+	}{
+		{
+			// 10 rewards accrued on each of 3 days
+			// Claimed shortly into the first day, 10 total claimed
+			name:                "claim the first day",
+			timeOffset:          time.Hour, // one hour into first window
+			initialAllocations:  []int64{10, 10, 10},
+			expectedAllocations: []int64{0, 10, 10},
+			initialClaimed:      100,
+			expectedClaimed:     100 + 10,
+			expectedNewRewards:  10,
+			initialClaimType:    types.CLAIM_DAILY,
+		},
+		{
+			// 10 rewards accrued on each of 3 days
+			// Claimed shortly into the second day, 20 total claimed
+			name:                "claim the second day",
+			timeOffset:          time.Hour * 25, // one hour into second window
+			initialAllocations:  []int64{10, 10, 10},
+			expectedAllocations: []int64{0, 0, 10},
+			initialClaimed:      100,
+			expectedClaimed:     100 + 20,
+			expectedNewRewards:  20,
+			initialClaimType:    types.CLAIM_DAILY,
+		},
+		{
+			// 10 rewards accrued on each of 3 days
+			// Claimed shortly into the second day, 20 total claimed
+			name:                "claim all days at once",
+			timeOffset:          time.Hour * 49, // one hour into third window
+			initialAllocations:  []int64{10, 10, 10},
+			expectedAllocations: []int64{0, 0, 0},
+			initialClaimed:      100,
+			expectedClaimed:     100 + 30,
+			expectedNewRewards:  30,
+			initialClaimType:    types.CLAIM_DAILY,
+		},
+		{
+			// Claimer already chose claim early
+			name:               "already chose to claim early",
+			timeOffset:         time.Hour,
+			initialAllocations: []int64{},
```

Comment:
Is `[]int64{}` the same as `[]int64{0,0,0,...}`? That's what I expect an early claimer's allocation array to look like.

Reply:
it's not the same, but the allocations don't really matter since the check on the claim type is upstream. I'll flush it out though just to make the test easier to interpret